utils/_context/_scenarios.py

```diff
@@ -840,6 +841,41 @@ def close_targets(self):
 
 
 class ParametricScenario(_Scenario):
+    class PersistentParametricTestConf(dict):
+        """ Parametric tests are executed in multiple thread, we need a mechanism to persist each parametrized_tests_metadata on a file"""
+
+        def __init__(self, outer_inst):
+            self.outer_inst = outer_inst
+            # To handle correctly we need to add data by default
+            self.update({"scenario": outer_inst.name})
+
+        def __setitem__(self, item, value):
+            super().__setitem__(item, value)
+            # Append to the file
+            with open(f"{self.outer_inst.host_log_folder}/context.json", "a") as f:
```

Comment:
Even if the probability is low, you can have both threads that want to write the file at the same moment.

It may better to write one file per thread, with the thread id in the file name, and then iterate on all file in deserialize

Reply:
you're right! The change is made!